be/src/util/jni_metrics.cpp

```diff
@@ -0,0 +1,122 @@
+// specific language governing permissions and limitations
+// under the License.
+
+#include "jni_metrics.h"
+
+#include <utility>
+
+#include "util/jni-util.h"
+#include "util/metrics.h"
+
+namespace doris {
+
+DEFINE_GAUGE_METRIC_PROTOTYPE_2ARG(jdbc_scan_connection_percent, MetricUnit::PERCENT);
+struct JdbcConnectionMetrics {
+    JdbcConnectionMetrics(std::shared_ptr<MetricEntity> entity) : entity(entity) {
+        INT_GAUGE_METRIC_REGISTER(entity, jdbc_scan_connection_percent);
+    }
+    void upate(int value) { jdbc_scan_connection_percent->set_value(value); }
+    std::shared_ptr<MetricEntity> entity;
+    IntGauge* jdbc_scan_connection_percent;
+};
+
+const char* JniMetrics::_s_hook_name = "jni_metrics";
+
+JniMetrics::JniMetrics(MetricRegistry* registry) {
+    DCHECK(registry != nullptr);
+    _registry = registry;
+    _server_entity = _registry->register_entity("server");
+    DCHECK(_server_entity != nullptr);
+    _server_entity->register_hook(_s_hook_name,
+                                  std::bind(&JniMetrics::update_jdbc_connection_metrics, this));
+    _init();
+}
+
+void JniMetrics::_init() {
+    JNIEnv* env = nullptr;
+    Status st = JniUtil::GetJNIEnv(&env);
+    if (!st.ok()) {
+        LOG(WARNING) << "get jni env failed. " << st.to_string();
+        return;
+    }
+    st = JniUtil::get_jni_scanner_class(env, "org/apache/doris/jdbc/JdbcDataSource",
+                                        &_jdbc_data_source_clz);
+    if (!st.ok()) {
+        LOG(WARNING) << "class org/apachhe/doris/jdbc/JdbcDataSource not find in jvm."
+                     << st.to_string();
+        return;
+    }
+    _get_connection_percent_id = env->GetStaticMethodID(
+            _jdbc_data_source_clz, "getConnectionPercent", "()Ljava/util/Map;");
+    st = JniUtil::GetJniExceptionMsg(env);
+    if (!st.ok()) {
+        LOG(WARNING) << "get getConnectionPercent method id failed. " << st.to_string();
+        return;
+    }
+    _is_init = true;
+    LOG(INFO) << "jni metrics inited successfully";
+}
+
+void JniMetrics::update_jdbc_connection_metrics() {
+    if (!_is_init) {
+        return;
+    }
+    JNIEnv* env = nullptr;
+    Status st = JniUtil::GetJNIEnv(&env);
+    if (!st.ok()) {
+        LOG(WARNING) << "get jni env failed. " << st.to_string();
+        return;
+    }
+    jobject metrics =
+            env->CallStaticObjectMethod(_jdbc_data_source_clz, _get_connection_percent_id);
+    std::map<std::string, std::string> result = JniUtil::convert_to_cpp_map(env, metrics);
+    st = JniUtil::GetJniExceptionMsg(env);
+    if (!st.ok()) {
+        LOG(WARNING) << "invoke getConnectionPercent method failed. " << st.to_string();
+        return;
+    }
+    for (auto item : result) {
+        std::string catalog_id = item.first;
+        int percent = std::stoi(item.second);
+        auto iter = _jdbc_connection_metrics.find(catalog_id);
+        if (iter == _jdbc_connection_metrics.end()) {
```

Comment:
if a catalog before exists,  but now not exists (  drop catalog xxx ) . Will `_jdbc_connection_metrics` always hold the object? Can I see the monitoring information of the deleted catalog?